### DISCUSSION TITLE
Best way to get the height of a renderable?

### DISCUSSION THREAD

**Discussion by attheodo** [Textualize/rich#2373] [Q&A]
I've implemented a pseudo-console panel inside my `Live` layout. It's based on a `Table.grid`.  As the user changes the terminal height or new items (rows) being added, they move beyond the viewport of the panel so to speak.

Is there a way I can grab dynamically the current height of the renderable so that I can do calculations and only show the appropriate portion of the rows?

**Comment by willmcgugan** [comment]
You can render it with `console.render_lines` which returns a list of segments for each line.

**Reply by attheodo** [reply]
@willmcgugan sorry not following, can you be a bit more verbose? Currently I have some events that when triggered I do something like this:
```
 self.layout[UIComponent.CONSOLE].update(self.console_panel.generate_panel())
 self.live.update(self.layout, refresh=True)
```

**Reply by attheodo** [reply]
@willmcgugan any chance you could provide some more detailed guidance here? 🙏

**Reply by willmcgugan** [reply]
The problem is that you can only know how the height of a renderable after you have rendered it. For instance you have to word-wrap a string before you know how many lines it is going to use.

We might be able to give you more concrete suggestions if we have a fully working example, and a better idea of what you want to implement.

**Reply by attheodo** [reply]
So I have a `Layout` that looks like this:

```python
 def make_layout(self) -> Layout:
        
        layout = Layout(name=UIComponent.APP)

        layout.split(
            Layout(name=UIComponent.CURRENT_CONTRACT, size=3),
            Layout(name=UIComponent.CMD_INPUT, size=3),
            Layout(name=UIComponent.ORDERS, ratio=2),
            Layout(name=UIComponent.POSITIONS, ratio=2),
            Layout(name=UIComponent.CONSOLE, ratio=1),
            Layout(name=UIComponent.FOOTER, size=3)
        )

        return layout
```

and is being rendered through `Live` like so while also accepting user input

```python
 def start_input_loop(self):
                    
        with Live(self.layout, auto_refresh=False, redirect_stderr=False, redirect_stdout=False, console=self.console) as live:
           
            self.live = live

            self.update_trades()
            self.load_current_positions()
           
            while True:
                
                while self.input_queue.qsize() > 0:
                    key = self.input_queue.get()
                    self.on_key_captured(key)
        
                IB.sleep(0.02)
```

Whenever the user enters a new command or whenever I want to "log" something to the console, I am basically updating a list and feeding it to `UIComponent.CONSOLE` that basically iterates the list of strings, creates a `Rich.Table` and calls a callback method in which I just do something like

```python
 def update_console(self, console_panel: ConsolePanel):
        
        self.layout[UIComponent.CONSOLE].update(self.console_panel.generate_panel())
        self.live.update(self.layout, refresh=True)
```

Now as "logged" items pile up the `Rich.Table` in the console `Panel` keeps getting more rows and the new ones get hidden below "the viewport" so to speak. What I am trying to achieve is get a sense of what is the actual size (height) of the console `Panel` inside the `Layout` so that I can dynamically discard the required `n` items from the items to be included at the console table so at any given point in time I only have the correct amount of rows in the table in order for all of them to be visible.

Hope this makes some more sense.
